driver/src/main/com/mongodb/reactivestreams/client/ObservableToPublisher.java

```diff
@@ -45,7 +47,13 @@ public void request(final long n) {
                                     + "Subscription.request(long n) MUST throw a java.lang.IllegalArgumentException if the "
                                     + "argument is <= 0."));
                         } else {
-                            subscription.request(n);
+                            try {
+                                subscription.request(n);
+                            } catch (MongoException e) {
+                                subscriber.onError(e);
+                            } catch (Exception e) {
```

Comment:
What's the argument for wrapping in MongoException?  

I was thinking it would just catch Throwable and return that directly to onError.